### PR TITLE
Fix fakeQuat memory leak

### PR DESCRIPTION
**Pull request by leHeix**
hi brother @ADRFranklin 

memory leaks everywhere in your code

fix

### CODE REVIEW

**Comment by ADRFranklin** [comment]
ah I didn't catch this one at the time apparently. Thanks.

File: src/Hooks.cpp [hunk 1]
```diff
@@ -625,9 +625,5 @@ void InstallPreHooks()
 {
 	std::memset(&Player::fakeHealth, 255, sizeof(Player::fakeHealth));
 	std::memset(&Player::fakeArmour, 255, sizeof(Player::fakeArmour));
-
-	for (int i = 0; i < 1000; i++)
-	{
-		Player::fakeQuat[i] = 0;
-	}
+	std::memset(&Player::fakeQuat[0], NULL, sizeof(Player::fakeQuat));
```
Comment (BigETI) [inline src/Hooks.cpp:628]: https://en.cppreference.com/w/cpp/algorithm/fill

File: src/Scripting.cpp [hunk 2]
```diff
@@ -234,9 +234,14 @@ static cell AMX_NATIVE_CALL SetFakeFacingAngle(AMX *amx, cell *params)
 	if(!IsPlayerConnected(playerid))
 		return 0;	
 
+	if (Player::fakeQuat[playerid])
```
Comment (BigETI) [inline src/Scripting.cpp:237]: This is not needed at all, deleting a null pointer is well defined.

**Review by BigETI — COMMENTED** [review]
You should generally avoid allocating each `glm::quat` into heap.

File: src/Hooks.cpp (hunk 1 above)
Comment (leHeix) [reply src/Hooks.cpp:628]: > You should generally avoid allocating each `glm::quat` into heap.

Sure, JustMichael can fix that. This pull request was made to fix a memory leak in SKY's current approach to store quaternions. I, however, agree with you.

> https://en.cppreference.com/w/cpp/algorithm/fill

memset is faster (or equally as efficient) than `std::fill`. They use memset 2 times above too, so I think bringing `std::fill` there just to, well, fill an array would be worthless when we're managing POD. If I were to use `std::fill`, I'd also use `std::array` and `std::array::fill`.

File: src/Scripting.cpp (hunk 2 above)
Comment (leHeix) [reply src/Scripting.cpp:237]: > This is not needed at all, deleting a null pointer is well defined.

You're right, that can be removed.

**Pull request merged**